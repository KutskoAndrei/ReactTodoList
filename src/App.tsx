import classes from './App.module.scss';
import ButtonPanel from './components/ButtonPanel/ButtonPanel';
import TodoList from './components/TodoList/TodoList';

function App() {

  return (
    <div className={classes.AppContent}>
      <div className={classes.controlPanel}>
        <ButtonPanel />
      </div>
      <div className={classes.contentPanel}>
        <TodoList />
      </div>
    </div>
  );
}

export default App;
