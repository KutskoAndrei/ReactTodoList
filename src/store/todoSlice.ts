import { FilterType } from "../models/FilterType";
import { ITodoItem } from "../models/ITodoItem";
import { createSlice, PayloadAction } from '@reduxjs/toolkit'

const todos: ITodoItem[] = [];
const filter: number = FilterType.ALL;

const todoSlice = createSlice({
    name: 'todos',
    initialState: {
        todos,
        filter
    },
    reducers: {
        addTodo(state, action: PayloadAction<ITodoItem>) {
            state.todos.push(action.payload);
        },
        removeTodo(state, action: PayloadAction<ITodoItem>) {
            state.todos = state.todos.filter((todo) => todo.id !== action.payload.id);
        },
        switchTodoStatus(state, action: PayloadAction<ITodoItem>) {
            const switchTodo = state.todos.find((todo) => todo.id = action.payload.id);
            if (switchTodo) switchTodo.checked = !switchTodo.checked;
        },
        updateTodos(state, action: PayloadAction<ITodoItem[]>) {
            state.todos = action.payload;
        },
        updateFilter(state, action: PayloadAction<FilterType>) {
            state.filter = action.payload;
        }
    }
});

export const {
    addTodo,
    removeTodo,
    switchTodoStatus,
    updateTodos,
    updateFilter
} = todoSlice.actions;

export default todoSlice.reducer;