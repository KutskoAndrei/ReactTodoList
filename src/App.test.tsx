import { RenderResult, fireEvent, render, screen } from '@testing-library/react';
import App from './App';
import { Provider } from 'react-redux';
import { store } from './store/store';

describe('App tests', () => {
  let component: RenderResult;

  beforeEach(() => {
    component = render(
      <Provider store={store}>
        <App />
      </Provider>
    )
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it('should render component', () => {
    const appComponentClass = component.container.getElementsByClassName('AppContent');
    expect(appComponentClass.length).toBe(1);
  });
});






// import React from 'react';
// import { render, screen } from '@testing-library/react';
// import App from './App';

// test('renders learn react link', () => {
//   render(<App />);
//   const linkElement = screen.getByText(/learn react/i);
//   expect(linkElement).toBeInTheDocument();
// });
