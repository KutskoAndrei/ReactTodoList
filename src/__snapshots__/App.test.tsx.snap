// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App tests should match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="AppContent"
      >
        <div
          class="controlPanel"
        >
          <div
            class="buttonPanel"
          >
            <div
              class="topButton"
            >
              <div
                class="regularButton"
                data-testid="addTodo"
              >
                <h3>
                  Add New Todo
                </h3>
              </div>
            </div>
            <div
              class="buttonPanelItem"
            >
              <div
                class="regularButton active"
                data-testid="getAllTodos"
              >
                <h3>
                  All Todos
                </h3>
              </div>
              <div
                class="scoreLabel"
              >
                <h1>
                  0
                </h1>
              </div>
            </div>
            <div
              class="buttonPanelItem"
            >
              <div
                class="regularButton"
                data-testid="getFinishedTodos"
              >
                <h3>
                  Finished Todos
                </h3>
              </div>
              <div
                class="scoreLabel"
              >
                <h1>
                  0
                </h1>
              </div>
            </div>
            <div
              class="buttonPanelItem"
            >
              <div
                class="regularButton"
                data-testid="getUnfinishedTodos"
              >
                <h3>
                  Unfinished Todos
                </h3>
              </div>
              <div
                class="scoreLabel"
              >
                <h1>
                  0
                </h1>
              </div>
            </div>
            <div
              class="bottomButton"
            >
              <div
                class="regularButton"
                data-testid="clearFinished"
              >
                <h3>
                  Clear Finished
                </h3>
              </div>
            </div>
          </div>
        </div>
        <div
          class="contentPanel"
        >
          <div
            class="todoList"
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="AppContent"
    >
      <div
        class="controlPanel"
      >
        <div
          class="buttonPanel"
        >
          <div
            class="topButton"
          >
            <div
              class="regularButton"
              data-testid="addTodo"
            >
              <h3>
                Add New Todo
              </h3>
            </div>
          </div>
          <div
            class="buttonPanelItem"
          >
            <div
              class="regularButton active"
              data-testid="getAllTodos"
            >
              <h3>
                All Todos
              </h3>
            </div>
            <div
              class="scoreLabel"
            >
              <h1>
                0
              </h1>
            </div>
          </div>
          <div
            class="buttonPanelItem"
          >
            <div
              class="regularButton"
              data-testid="getFinishedTodos"
            >
              <h3>
                Finished Todos
              </h3>
            </div>
            <div
              class="scoreLabel"
            >
              <h1>
                0
              </h1>
            </div>
          </div>
          <div
            class="buttonPanelItem"
          >
            <div
              class="regularButton"
              data-testid="getUnfinishedTodos"
            >
              <h3>
                Unfinished Todos
              </h3>
            </div>
            <div
              class="scoreLabel"
            >
              <h1>
                0
              </h1>
            </div>
          </div>
          <div
            class="bottomButton"
          >
            <div
              class="regularButton"
              data-testid="clearFinished"
            >
              <h3>
                Clear Finished
              </h3>
            </div>
          </div>
        </div>
      </div>
      <div
        class="contentPanel"
      >
        <div
          class="todoList"
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
